web/js/codeworld.js

```diff
@@ -221,6 +221,59 @@ function initCodeworld() {
         }
     });
     window.codeworldEditor.refresh();
+    window.codeworldEditor.on("cursorActivity", function() {
+        const prevDiv = document.getElementById("function-details");
+        if (prevDiv) prevDiv.remove();
+
+        const enclosingFunction = getEnclosingFunction(window.codeworldEditor);
+
+        if (!enclosingFunction || !enclosingFunction.string ||  !window.codeWorldSymbols[enclosingFunction.string]) {
+            return;
+        }
+        const cursor = window.codeworldEditor.getCursor();
+        const currentToken = window.codeworldEditor.getTokenAt(cursor);
+        const lineTokens = window.codeworldEditor.getLineTokens(cursor.line);
+        const argIndex =
+            lineTokens.filter(
+                elem =>
+                    getEnclosingFunction(
+                        window.codeworldEditor,
+                        {
+                            line: cursor.line, ch: elem.end
+                        })
+                        .string === enclosingFunction.string
+                    && elem.type && elem.type !== "keyword"
+                    && elem.type!=="bracket-0"
+                    && elem.type !== "built-in"
+            ).findIndex(elem => (elem.start === currentToken.start || elem.end === currentToken.start || elem.start === currentToken.end || elem.end === currentToken.end));
+        const topDiv = document.createElement('div');
+
+        topDiv.title = enclosingFunction.string;
+        topDiv.id = "function-details";
+        const keywordData = window.codeWorldSymbols[enclosingFunction.string];
+
+        const docDiv = document.createElement('div');
+        docDiv.classList.add("function-tooltip-styling");
+
+        const annotation = document.createElement('div');
+        renderDeclaration(annotation, enclosingFunction.string, keywordData, 9999, argIndex);
+        annotation.className = 'hover-decl';
+        docDiv.appendChild(annotation);
+
+        topDiv.appendChild(docDiv);
+        window.codeworldEditor.addWidget(cursor, topDiv, true, "above", "left");
+    });
+
+    window.getEnclosingFunction = (cm, cur = cm.getCursor()) => {
+        const token = cm.getTokenAt(cur);
+        const line = cm.getLine(cur.line);
+        if (! line.includes('(')) return;
```

Comment:
I think you meant `line.substr(0, token.end).includes(...)` here?  Otherwise you might continue, but later fail to find the parenthesis.

Reply:
This isn't required after implementing your approach.